src/extremeweatherbench/data_loader.py

```diff
@@ -0,0 +1,169 @@
+    will be inferred from the provided source (in this case, s3). Otherwise,
+    the filesystem will assumed to be local.
+    Args:
+        eval_config: The evaluation configuration.
+        forecast_schema_config: The forecast schema configuration.
+
+    Returns:
+        The opened forecast dataset.
+    """
+    logger.debug("Opening forecast dataset")
+    if "zarr" in eval_config.forecast_dir:
+        forecast_ds = xr.open_zarr(eval_config.forecast_dir, chunks="auto")
+    elif (
+        "parq" in eval_config.forecast_dir
+        or "json" in eval_config.forecast_dir
+        or "parquet" in eval_config.forecast_dir
+    ):
+        forecast_ds = open_kerchunk_reference(eval_config)
+    else:
+        raise TypeError(
+            "Unknown file type found in forecast path, only json, parquet, and zarr are supported."
+        )
+    forecast_ds = _rename_forecast_dataset(forecast_ds, forecast_schema_config)
+    forecast_ds = utils.maybe_convert_dataset_lead_time_to_int(forecast_ds)
+
+    return forecast_ds
+
+
+def open_kerchunk_reference(
+    eval_config: config.Config,
+) -> xr.Dataset:
+    """Open a dataset from a kerchunked reference file in parquet or json format.
+    This has been built for the CIRA MLWP S3 bucket's data (https://registry.opendata.aws/aiwp/),
+    but can work with other data in the future. Currently only supports CIRA data unless
+    schema is identical to the CIRA schema.
+
+    Args:
+        file: The path to the kerchunked reference file.
+        remote_protocol: The remote protocol to use.
+
+    Returns:
+        The opened dataset.
+    """
+    if "parq" in eval_config.forecast_dir or "parquet" in eval_config.forecast_dir:
+        storage_options = {
+            "remote_protocol": eval_config.remote_protocol,
+            "remote_options": {"anon": True},
+        }  # options passed to fsspec
+        open_dataset_options: dict = {"chunks": {}}  # opens passed to xarray
+
+        kerchunk_ds = xr.open_dataset(
+            eval_config.forecast_dir,
+            engine="kerchunk",
+            storage_options=storage_options,
+            open_dataset_options=open_dataset_options,
+        )
+        kerchunk_ds = kerchunk_ds.compute()
+    elif "json" in eval_config.forecast_dir:
+        kerchunk_ds = xr.open_dataset(
+            "reference://",
+            engine="zarr",
+            backend_kwargs={
+                "consolidated": False,
+                "storage_options": {
+                    "fo": eval_config.forecast_dir,
+                    "remote_protocol": eval_config.remote_protocol,
+                    "remote_options": {"anon": True},
+                },
+            },
+        )
+    else:
+        raise TypeError(
+            "Unknown kerchunk file type found in forecast path, only json and parquet are supported."
+        )
+    if eval_config.forecast_source == "cira":
+        kerchunk_ds = kerchunk_ds.rename({"time": "lead_time"})
```

Comment:
See prior comments about hard-coding CIRA info in the config. It's okay to have some helper functions on the side, but this should not be in the public interfaces.